packages/react/src/SelectPanel/SelectPanel.tsx

```diff
@@ -17,11 +17,97 @@ import type {FocusZoneHookSettings} from '../hooks/useFocusZone'
 import {useId} from '../hooks/useId'
 import {useProvidedStateOrCreate} from '../hooks/useProvidedStateOrCreate'
 import {LiveRegion, LiveRegionOutlet, Message} from '../internal/components/LiveRegion'
+import useSafeTimeout from '../hooks/useSafeTimeout'
+import type {FilteredActionListLoadingType} from '../FilteredActionList/FilteredActionListLoaders'
+import {FilteredActionListLoadingTypes} from '../FilteredActionList/FilteredActionListLoaders'
 import {useFeatureFlag} from '../FeatureFlags'
+import {announce} from '@primer/live-region-element'
 
 import classes from './SelectPanel.module.css'
 import {clsx} from 'clsx'
 
+// we add a delay so that it does not interrupt default screen reader announcement and queues after it
+const delayMs = 500
+const loadingDelayMs = 1000
+
+const getItemWithActiveDescendant = (
+  listRef: React.RefObject<HTMLElement>,
+  items: FilteredActionListProps['items'],
+) => {
+  const listElement = listRef.current
+  const activeItemElement = listElement?.querySelector('[data-is-active-descendant]')
+
+  if (!listElement || !activeItemElement?.textContent) return
+
+  const optionElements = listElement.querySelectorAll('[role="option"]')
+
+  const index = Array.from(optionElements).indexOf(activeItemElement)
+  const activeItem = items[index] as ItemInput | undefined
+
+  const text = activeItem?.text
+  const selected = activeItem?.selected
+
+  return {index, text, selected}
+}
+
+function announceFilterFocused() {
+  const liveRegion = document.querySelector('live-region')
+
+  liveRegion?.clear() // clear previous announcements
+
+  announce('Focus on filter text box and list of items', {
+    delayMs,
+    from: liveRegion ? liveRegion : undefined, // announce will create a liveRegion if it doesn't find one
+  })
+}
+
+function announceNoItems() {
+  const liveRegion = document.querySelector('live-region')
+
+  liveRegion?.clear() // clear previous announcements
+
+  announce('No matching items.', {
+    delayMs,
+    from: liveRegion ? liveRegion : undefined, // announce will create a liveRegion if it doesn't find one
+  })
+}
+
+function announceLoading() {
+  const liveRegion = document.querySelector('live-region')
+
+  liveRegion?.clear() // clear previous announcements
+
+  announce('Loading.', {
+    delayMs,
+    from: liveRegion ? liveRegion : undefined, // announce will create a liveRegion if it doesn't find one
+  })
+}
```

Comment:
these are pretty similar, could probably be combined into a single function that takes the announce text as paramenter